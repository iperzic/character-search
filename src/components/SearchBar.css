.SearchBar {
    margin-bottom: 30px;
    padding: 30px;
    color: white;
    background-color: #000;
}

.SearchBar__input {
    font-size: 20px;
}
