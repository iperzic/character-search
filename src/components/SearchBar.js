import React from 'react';
import PropTypes from 'prop-types';

import './SearchBar.css';

const SearchBar = ({ onQueryChange }) => (
  <div className="SearchBar">
    <input type="search" placeholder="Search Characters" onChange={onQueryChange} />
  </div>
);

SearchBar.propTypes = {
  onQueryChange: PropTypes.func.isRequired,
};

export default SearchBar;
