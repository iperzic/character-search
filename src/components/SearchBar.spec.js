import React from 'react';
import { shallow } from 'enzyme';

import SearchBar from './SearchBar';

describe('SearchBar', () => {
  let component;
  let queryChangeSpy;

  beforeEach(() => {
    queryChangeSpy = jest.fn();
    component = shallow(<SearchBar onQueryChange={queryChangeSpy} />);
  });

  it('should show an input for query', () => {
    expect(component.getElement()).toMatchSnapshot();
  });

  describe('when the input is changed', () => {
    it('should call the onQueryChange prop with the query', () => {
      const name = 'character name';
      const queryInput = component.find('.SearchBar__input');

      queryInput.simulate('change', {
        currentTarget: {
          value: name,
        },
      });

      expect(queryChangeSpy).toHaveBeenCalledWith({ currentTarget: { value: name } });
    });
  });
});
