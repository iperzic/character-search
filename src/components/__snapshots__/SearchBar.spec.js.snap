// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar should show an input for query 1`] = `
<div
  className="SearchBar"
>
  <input
    className="SearchBar__input"
    onChange={[Function]}
    placeholder="Search Characters"
    type="search"
  />
</div>
`;
